crates/api_common/src/utils.rs

```diff
@@ -731,12 +731,40 @@ pub fn create_login_cookie(jwt: Sensitive<String>) -> Cookie<'static> {
   cookie
 }
 
+/// Ensure that ban/block expiry is in valid range. If its in past, throw error. If its more
+/// than 10 years in future, convert to permanent ban. Otherwise return the same value.
+pub fn check_expire_time(expires_unix_opt: Option<i64>) -> LemmyResult<Option<DateTime<Utc>>> {
+  if let Some(expires_unix) = expires_unix_opt {
+    let expires = Utc
+      .timestamp_opt(expires_unix, 0)
+      .single()
+      .ok_or(LemmyErrorType::InvalidUnixTime)?;
```

Comment:
Unix time as a simple number works much better IMO than trying to deal with string date formats. I think those expires are the only "date" info currently in the API / api_common. I'm not completely against changing it, but unix time seems much easier and universal to use in an API than formatted date strings.

Reply:
I mean we are already sending out datetime strings for all the created and updated fields, so it shouldnt be a problem to use it for receiving too. But no big deal.

Reply:
I still think a simple int is best, because its going to be universally supported by all programming languages and their api clients, and the typings generated are going to be simple.